homeassistant/components/fmi/config_flow.py

```diff
@@ -0,0 +1,131 @@
+"""Config flow for FMI (Finnish Meteorological Institute) integration."""
+
+import fmi_weather_client as fmi_client
+from fmi_weather_client.errors import ClientError, ServerError
+import voluptuous as vol
+
+from homeassistant import config_entries, core
+from homeassistant.const import CONF_LATITUDE, CONF_LONGITUDE, CONF_NAME, CONF_OFFSET
+from homeassistant.core import callback
+from homeassistant.helpers import config_validation as cv
+
+from . import base_unique_id
+from .const import _LOGGER, FORECAST_OFFSET
+
+
+async def validate_user_config(hass: core.HomeAssistant, data):
+    """Validate input configuration for FMI.
+
+    Data contains Latitude / Longitude provided by user or from
+    HASS default configuration.
+    """
+    latitude = data[CONF_LATITUDE]
+    longitude = data[CONF_LONGITUDE]
+
+    errors = ""
+
+    # Current Weather
+    try:
+        weather_data = await hass.async_add_executor_job(
+            fmi_client.weather_by_coordinates, latitude, longitude
+        )
+
+        return {"place": weather_data.place, "err": ""}
```

Comment:
Instead of trying to return a dictionary to pass an error along, raise in case of an error. This can be catched downstream